### PR TITLE
Improve naming of ReadStdin

### PR DESCRIPTION
**Pull request by soc**
We don't need to say everything twice, methods are already prefixed
with “read”.

### CODE REVIEW

**Comment by soc** [comment]
No deprecation, because the original change happened earlier in 2.11, so no stable version with `ReadStdin` is out in the wild.

**Comment by Ichoran** [comment]
LGTM.

File: src/library/scala/io/StdIn.scala [hunk 1]
```diff
@@ -225,4 +225,4 @@ private[scala] trait ReadStdin {
   }
 }
 
-object ReadStdin extends ReadStdin { }
+object StdIn extends StdIn
```
Comment (retronym) [inline src/library/scala/io/StdIn.scala:228]: This change isn't source compatible.

Comment (soc) [inline src/library/scala/io/StdIn.scala:228]: Source compatible with what version? Should I introduce a deprecation for a milestone? ... Not sure what's the suggested action here.

**Comment by retronym** [comment]
Oh sorry, I missed your original comment about that.

LGTM

**Pull request merged**